x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/index/IndexResolution.java

```diff
@@ -12,22 +12,28 @@
 import java.util.Collections;
 import java.util.Map;
 import java.util.Objects;
+import java.util.Set;
 
 public final class IndexResolution {
 
-    public static IndexResolution valid(EsIndex index, Map<String, FieldCapabilitiesFailure> unavailableClusters) {
+    public static IndexResolution valid(
+        EsIndex index,
+        Set<String> resolvedIndices,
+        Map<String, FieldCapabilitiesFailure> unavailableClusters
+    ) {
         Objects.requireNonNull(index, "index must not be null if it was found");
+        Objects.requireNonNull(resolvedIndices, "resolvedIndices must not be null");
         Objects.requireNonNull(unavailableClusters, "unavailableClusters must not be null");
-        return new IndexResolution(index, null, unavailableClusters);
+        return new IndexResolution(index, null, resolvedIndices, unavailableClusters);
     }
 
-    public static IndexResolution valid(EsIndex index) {
-        return valid(index, Collections.emptyMap());
+    public static IndexResolution valid(EsIndex index, Set<String> resolvedIndices) {
```

Comment:
Nit: to avoid the noise in the PR, keep the old method in place and extract the name from EsIndex:
```
valid(EsIndex index) { return valid(index,  singletonSet(index.name())}
```

Reply:
Good idea. Except we should use index.concreteIndices(), not index name, since the latter is the original unresolved user requested index expression. Fixed in next push.